tests/Dompdf/Tests/DompdfTest.php

```diff
@@ -98,4 +98,125 @@ public function testSpaceAtStartOfSecondInlineTag()
         $this->assertEquals("one", $text_frame_contents[0]);
         $this->assertEquals(" - two", $text_frame_contents[1]);
     }
+
+    private function _pageBreak($orphans, $widows, $lines_in_paragraph, $lines_that_could_fit_on_first_page)
+    {
+        $dompdf = new Dompdf();
+
+        $callback_count = 0;
+        $number_of_lines_on_second_page = 0;
+
+        // Use a callback to inspect the frame tree; otherwise FrameReflower\Page::reflow()
+        // will dispose of it before dompdf->render finishes
+        $dompdf->setCallbacks(array('test' => array(
+            'event' => 'end_page_render',
+            'f' => function($params) use (&$number_of_lines_on_second_page, &$callback_count) {
+                if ($callback_count == 1) {
+                    $frame = $params["frame"];
+                    foreach ($frame->get_children() as $child) {
+                        foreach ($child->get_children() as $grandchild) {
+                            if ($grandchild->get_node()->nodeName == "#text")
+                                $number_of_lines_on_second_page++;
+                        }
+                    }
+                }
+                $callback_count++;
+            }
+        )));
+
+        $html = "<p>";
+        $filler = array();
+        $MAX_FILLER = 47; // Is there a better way to know how many lines fit on a page?
```

Comment:
Using a fixed, and smaller page size together with a fixed line height should make this easier to handle.